manim/mobject/number_line.py

```diff
@@ -15,41 +57,90 @@
 from ..utils.config_ops import merge_dicts_recursively
 from ..utils.simple_functions import fdiv
 from ..utils.space_ops import normalize
-from ..utils.color import LIGHT_GREY
+from ..utils.color import Colors, LIGHT_GREY
 
 
 class NumberLine(Line):
+    """Creates a number line"""
+
     def __init__(
         self,
-        color=LIGHT_GREY,
-        unit_size=1,
-        width=None,
-        include_ticks=True,
-        tick_size=0.1,
-        tick_frequency=1,
+        color: Colors = LIGHT_GREY,
+        unit_size: int = 1,
+        width: float = None,
+        include_ticks: bool = True,
+        tick_size: float = 0.1,
+        tick_frequency: int = 1,
         # Defaults to value near x_min s.t. 0 is a tick
         # TODO, rename this
-        leftmost_tick=None,
+        leftmost_tick: float = None,
         # Change name
-        numbers_with_elongated_ticks=[0],
-        include_numbers=False,
-        numbers_to_show=None,
-        longer_tick_multiple=2,
-        number_at_center=0,
-        number_scale_val=0.75,
-        label_direction=DOWN,
-        line_to_number_buff=MED_SMALL_BUFF,
-        include_tip=False,
-        tip_width=0.25,
-        tip_height=0.25,
-        add_start=0,  # extend number line by this amount at its starting point
-        add_end=0,  # extend number line by this amount at its end point
-        decimal_number_config={"num_decimal_places": 0},
-        exclude_zero_from_default_numbers=False,
-        x_min=-config["frame_x_radius"],
-        x_max=config["frame_x_radius"],
+        numbers_with_elongated_ticks: typing.List[float] = [0],
+        include_numbers: bool = False,
+        numbers_to_show: typing.List[float] = None,
+        longer_tick_multiple: int = 2,
+        number_at_center: float = 0,
+        number_scale_val: float = 0.75,
+        label_direction: np.ndarray = DOWN,
+        line_to_number_buff: float = MED_SMALL_BUFF,
+        include_tip: bool = False,
+        tip_width: float = 0.25,
+        tip_height: float = 0.25,
+        add_start: float = 0,  # extend number line by this amount at its starting point
+        add_end: float = 0,  # extend number line by this amount at its end point
+        decimal_number_config: dict = {"num_decimal_places": 0},
```

Comment:
```suggestion
        decimal_number_config: typing.Dict[str,int] = {"num_decimal_places": 0},
```

Reply:
actually, it is not appropriate in this context because  decimal_number_config is essentially the kwargs for DecimalNumber class and the values for the dictionary  entries passed could other types.

Maybe `Any` instead of `int` is appropriate.

That said, I think this is too confusing and somewhere stems from the overuse of passing 'dictionaries' in manim. 

The correct way should be to define a dataclass and then pass it around. ... nevertheless whatever we do writing scalable and maintainable code in python is always going to be tricky in this project or any other project.

Given all this ambiguity, my humble suggestion is to leave it at `dict`, and when (and if) you refactor the manim code to use proper types (instead of dictionaries) then the types would help otherwise this does not bring any value.